src/simulators/statevector/statevector_state.hpp

```diff
@@ -578,16 +595,27 @@ void State<statevec_t>::snapshot_pauli_expval(const Operations::Op &op,
   }
   // add to snapshot
   Utils::chop_inplace(expval, json_chop_threshold_);
-  data.add_average_snapshot("expectation_value", op.string_params[0],
-                            BaseState::creg_.memory_hex(), expval, variance);
+  switch (type) {
+    case SnapshotDataType::average:
+      data.add_average_snapshot("expectation_value", op.string_params[0],
+                            BaseState::creg_.memory_hex(), expval, false);
```

Comment:
What is the last `true` and `false` argument? I'd definitely put an enum here so I don't have to go the function definition to know what's this boolean means.